A29-xds-tls-security.md

```diff
@@ -145,31 +144,42 @@ Use of XdsServerCredentials:
+or XdsServerCredentials is in force. If validation fails, the update is NACKed. When an update is
+accepted the correponding security configuration in CDS or LDS is used if the corresponding
+Xds*Credentials is in force.
 
 #### CDS for Client Side Security
 
 The CDS policy is the top level LB policy applied to all the connections under that policy.
 [A27:CDS][] describes the flow. The field [`transport_socket`][CL-TS] is used to extract the
 [`UpstreamTlsContext`][UTC] as described [here][CL-TS-comment].
 Note that we don't (currently) support [`transport_socket_matches`][CL-TS-matches].
-The `UpstreamTlsContext` thus obtained is passed down to all the child policies and connections.
-How this is done is language dependent and is similar to how the implementations pass policy
-information down to child policies.
+The security configuration extracted from the `UpstreamTlsContext` thus obtained
+is passed down to all the child policies and connections. How this is done is language dependent
+and is similar to how the implementations pass policy information down to child policies.
+For example, C-core uses channel args to pass security configuration down, whereas
+Java constructs a dynamic `SslContextProvider` that is directly usable by a sub-channel's
+TLS handshaker to build an `SslContext` for the pending TLS handshake.
 
-[`common_tls_context`][CTC] in the `UpstreamTlsContext` contains the required security configuration.
-See below for [`CommonTlsContext`][CTC-type] processing.
+[`common_tls_context`][CTC] in the `UpstreamTlsContext` contains the required security
+configuration. See below for [`CommonTlsContext`][CTC-type] processing.
 
 A secure client at a minimum requires [`validation_context_certificate_provider_instance`][VCCPI1]
 inside [`combined_validation_context`][CVC] to be able to validate the server certificate
-in the TLS mode. If this field is not set, gRPC will NACK the CDS update. The client identity
-certificate is configured via the [`tls_certificate_certificate_provider_instance`][TCCPI]
-field which is optional. The client certificate is sent to the server if the server requests
-or requires it in the TLS handshake. If the server requires the client certificate but is not
-present then the TLS handshake fails.
+in the TLS mode. As part of validating all CDS updates, if this field is not set in a CDS,
+gRPC will NACK the CDS update. The client identity certificate is configured via the
+[`tls_certificate_certificate_provider_instance`][TCCPI] field which is optional. The
+client certificate is sent to the server if the server requests or requires it in the
+TLS handshake. If the server requires the client certificate but is not
+configured on the client side then the TLS handshake will fail because of the client's inability
+to send the certificate.
 
-If any of the following fields are present, gRPC will NACK the CDS update:
+As part of validating all CDS updates, if any of the following fields are present, gRPC
```

Comment:
Eric and I talked a bit about this: we can ignore a field (instead of NACKint it as unsupported) if that doesn't make the implementation any more insecure. Will re-evaluate the list with that goal.